fili-core/src/main/java/com/yahoo/bard/webservice/web/FilteredThetaSketchMetricsHelper.java

```diff
@@ -439,11 +466,12 @@ public String generateMetricName(String filterString) {
 
     @Override
     public Set<Dimension> gatherFilterDimensions(Filter filter, Set<Dimension> dimensions) {
-
         if (filter instanceof SelectorFilter) {
             dimensions.add(((SelectorFilter) filter).getDimension());
+        } else if (filter instanceof InFilter) {
```

Comment:
Blegh. Seems to me like this is something that we should pull onto a method
on the Filter interface. A few seconds of poking suggests this could be gnarly
though, so I'm not going to block on it.

But we're probably going to have this problem come up again.

Reply:
The Helper class itself is the first code smell.  The if else chain is the second. Yeah.